apps/public-docsite-v9/src/Concepts/Slots/Slots.stories.mdx

```diff
@@ -0,0 +1,379 @@
+import { Meta, Story } from '@storybook/addon-docs';
+import { SlotSketch } from './SlotSketch.stories';
+
+<Meta title="Concepts/Developer/Customizing Components with Slots" />
+
+## Customizing Components with Slots
+
+Fluent UI React components have parts that are designed to be modified or replaced.
+These are called slots.
+Slots provide a more flexible approach over render callbacks used in previous versions of Fluent UI
+and are conceptually similar to slots in other component libraries and frameworks.
+
+- Each slot is exposed as a top-level prop of the same name.
+- Some slots have default content and others are empty by default.
+- Slots may target different types of elements or components to restrict the type of content.
+- You can fill a slot with a primitive value, JSX/TSX, props objects, or a render function.
+
+You can think of component slots like a template where the dotted line parts can be cut out and replaced.
+For example, the `Input` component has `contentBefore` and `contentAfter` slots.
+
+<SlotSketch />
+
+### When to use slots
+
+Use a component's slots when you want to
+
+- set the content of a component part
+- customize the style of a component part (via `className`)
+- customize the props passed to a component part
+- subscribe to event handlers of a component part
+- change the element type of a component part (via `as`)
+- completely replace existing content of a slot
+
+Later in this topic, you will find examples covering each of these scenarios.
+
+### When to not use slots
+
+- If you are trying to change the visual appearance every instance of a component, prefer to customize the theme.
+  Fluent UI React components leverage theme design tokens to render consistently.
+  For example, if you want to have borders to be a specific color on `:hover`, you can create a theme that sets `colorNeutralStroke1Hover`.
+
+- If you need to slightly adjust the display of a specific component instance, prefer to apply a custom style.
+  For example, if you want to make a particular component have a thicker border.
+  You create the style using `makeStyles` and/or `mergeClasses` and then apply it to the component using `className`.
```

Comment:
This paragraph is a bit unclear to me as it's under "When to not use slots".

```tsx
const useClasses = makeStyles({
  // Option A
  root: { [`.${slotClassName}`]: { color: 'red' } },
  // Option B
  slot: { color: 'red' },
});

// Option A
<Component className={classes.root} />

// Option B
<Component slot={{ className: classes.slot }}>
```

Is it a recommendation to use "Option A"?

Reply:
I'm talking about Option A here.  I'll mention the styling components topic to clarity.

Reply:
I don't think that "Option A" is a recommended approach actually. "Option B" is preferred.

Can we change/remove this to reflect this?

Reply:
If a developer want to change a button border, they would apply the className to the Button right? (option A).  If they want to customize the icon border, they would apply the className to the slot (option B), which would be using slots by passing the classname prop to the slot.

Reply:
Correct, but please check the snippet: it contains a nested selector in Option A that applies styles a slot.